module/x/gravity/keeper/pool.go

```diff
@@ -265,7 +287,7 @@ func (k Keeper) GetBatchFeeByTokenType(ctx sdk.Context, tokenContractAddr string
 		}
 		batchFee.TotalFees = batchFee.TotalFees.Add(fee.Amount)
 		txCount += 1
-		return txCount < int(maxElements)
+		return txCount == int(maxElements)
```

Comment:
if I'm interpreting this right the closure exits when the function we pass it returns false, so == will return false immediately and not give us the correct amount in fees.

Reply:
It `break`s when we return true:

```
if cb(iter.Key(), &transact) {
	break
}
```
We want to return false until we're done iterating

Reply:
ah in that case I was confused when I wrote this and it's a good correction.

Thanks!